xmtp/src/conversations.rs

```diff
@@ -304,12 +465,11 @@ where
             content_bytes: message.content.clone(),
         };
         let olm_message = session.encrypt(&payload.encode_to_vec());
-        let ciphertext = match olm_message {
-            OlmMessage::Normal(message) => message.to_bytes(),
-            OlmMessage::PreKey(prekey_message) => prekey_message.to_bytes(),
-        };
 
-        let envelope = PadlockMessageEnvelope {
+        let ciphertext =
+            serde_json::to_vec(&olm_message).map_err(|_| ConversationError::Unknown)?;
```

Comment:
Uses the built in serializer for the OlmMessage rather manually handling each type.

Reply:
The nice thing about `serde_json::to_vec` is that it doesn't require an additional field to distinguish pre-key and non-pre-key messages. The bad thing is that it will serialize it in a much bulkier format (JSON string) rather than `to_bytes` which actually uses the protobuf representation.

I was thinking I should actually update the invite handling to use `to_bytes` for more compact messages.

Reply:
I waffled on this one. for the same reasons. I'll leave the encodings as they are for now then. And we can look at changing the Invite Protos later then